configuration_version_integration_test.go

```diff
@@ -242,6 +242,73 @@ func TestConfigurationVersionsArchive(t *testing.T) {
 	client := testClient(t)
 	ctx := context.Background()
 
+	w, wCleanup := createWorkspace(t, client, nil)
+	defer wCleanup()
+
+	cv, cvCleanup := createConfigurationVersion(t, client, w)
+	defer cvCleanup()
+
+	t.Run("when the configuration version exists and has been uploaded", func(t *testing.T) {
+		err := client.ConfigurationVersions.Upload(
+			ctx,
+			cv.UploadURL,
+			"test-fixtures/config-version",
+		)
+		require.NoError(t, err)
+
+		// We do this is a small loop, because it can take a second
+		// before the upload is finished.
+		for i := 0; ; i++ {
+			refreshed, err := client.ConfigurationVersions.Read(ctx, cv.ID)
+			require.NoError(t, err)
+
+			if refreshed.Status == ConfigurationUploaded {
+				break
+			}
+
+			if i > 10 {
+				t.Fatal("Timeout waiting for the configuration version to be uploaded")
+			}
+
+			time.Sleep(1 * time.Second)
+		}
+
+		// configuration version should not be achievable, since it's the latest version
+		err = client.ConfigurationVersions.Archive(ctx, cv.ID)
+		assert.Error(t, err)
+		assert.EqualError(t, err, "transition not allowed")
+
+		// create subsequent version, since the latest configuration version cannot be archived
+		newCv, newCvCleanup := createConfigurationVersion(t, client, w)
+		err = client.ConfigurationVersions.Upload(
+			ctx,
+			newCv.UploadURL,
+			"test-fixtures/config-version",
+		)
+		require.NoError(t, err)
+		defer newCvCleanup()
```

Comment:
I think we might want a similar loop here to what is done for the first uploaded configuration version for similar reasons (e.g., to make sure we attempt the archive only after it is actually uploaded).

Reply:
Good catch 👍 Added